service/frontend/domainHandler.go

```diff
@@ -77,33 +76,40 @@ func (d *domainHandlerImpl) registerDomain(ctx context.Context,
 		return err
 	}
 
-	clusterMetadata := d.clusterMetadata
-	// TODO remove the IsGlobalDomainEnabled check once cross DC is public
-	if clusterMetadata.IsGlobalDomainEnabled() && !clusterMetadata.IsMasterCluster() {
-		return errNotMasterCluster
+	if registerRequest.GetName() == "" {
+		return errDomainNotSet
 	}
-	if !clusterMetadata.IsGlobalDomainEnabled() {
+
+	if !d.clusterMetadata.IsGlobalDomainEnabled() {
+		registerRequest.IsGlobalDomain = common.BoolPtr(false)
```

Comment:
I would return an error if a customer want a global domain which is not supported, rather than silently change it to local domain.